server/controllers/board.js

```diff
@@ -0,0 +1,119 @@
+const Board = require("../models/Board");
+const Card = require("../models/Card");
+const asyncHandler = require("express-async-handler");
+const BoardColumn = require("../models/BoardColumn");
+
+// @route POST /users
+// @desc Search for users
+// @access Private
+exports.listBoard = asyncHandler(async (req, res, next) => {
+  
+  const board = await Board.find({
+    user: req.user.id
+  }).populate({
+    path: 'columns',
+    populate: {
+      path: 'cards'
+    }
+  });
+
+  if(!board) {
+    throw new Error("No board found with given id");
```

Comment:
Great that you noticed async-handler will deal with the json response. However, this would show as a 500 error so best to set the response status to 404 just so we're following good HTTP principles.

Thanks!